src/Toolkit/Toolkit/UI/Controls/PopupViewer/PopupViewer.cs

```diff
@@ -185,7 +216,7 @@ private void OnPopupPropertyChanged(Popup? oldPopup, Popup? newPopup)
             }
             if (GetTemplateChild("SubFrameView") is NavigationSubView subView)
             {
-                subView.Navigate(content: Popup, true);
+                _ = subView.Navigate(content: Popup, skipRaisingEvent: true);
```

Comment:
I am finding that this throws when closing the popup (i.e. when Popup is null) unless I tell Navigate to clear the navigation stack.

```suggestion
                _ = subView.Navigate(content: Popup, skipRaisingEvent: true, clearNavigationStack: (Popup == null));
```